proto/p4/config/p4info.proto

```diff
@@ -20,6 +20,38 @@ import "google/protobuf/any.proto";
 
 package p4.config;
 
+message Documentation {
+  // A brief description of something, e.g. one sentence
+  string brief = 1;
+  // A more verbose description of something. Multiline is accepted. Markup format (if any) is TBD.
+  string description = 2;
+}
+
+// Top-level package documentation describing the forwarding pipeline config
+// Can be used to manage multiple P4 packages.
+message PkgInfo {
+    // a definitive name for this configuration, e.g. switch.p4_v1.0
+    string name = 1;
+    // configuration version, free-format string
+    string version = 2;
+    // brief and detailed descriptions
+    Documentation doc = 3;
+    // Miscellaneous metadata, free-form; a way to extend PkgInfo
+    repeated string annotations = 4;
+    // date the configuration was built or compiled
+    string build_date = 5;
+    // the toolchain used to build it, e.g. "p4c-bm2-ss v 0.0.5"
+    string compiler = 6;
+    // the target architecture, e.g. "bmv2 simple-switch"
+    string target = 7;
```

Comment:
Upon opening the p4.org home page, I see words promoting the target independence of P4.  Is the target field an attempt to limit a P4 program's usage to a specific target?

Reply:
Practically speaking most P4 programs will only run on a specific target once they are compiled with a specific backend, The P4Info isn't intented to "limit" anything, it is conveying things which were already decided by the code and compiler. This field will likely get populated by the compiler or some post-processing tool during make. On the other hand, if the P4 code was written for a specific target (e.g. using vendor-specific directives) the P4 source might even contain metadata use to populate this field.

Reply:
I will be doing a full review of this proposal very soon but regarding this point I tend to agree with @teverman. P4Info is not produced by a specific backend. As a matter of fact it is possible to generate the P4Info without running any real backend (by using the p4test compiler binary). I believe the P4Info should stay exactly the same independently of the **target**. However, it seems reasonable to change this field to `string arch`, as a P4 program is indeed written for a specific **architecture**. The idea is that the PSA architecture will be supported by all switch targets, while each target may support an additional vendor-specific architecture. I do think it is a very good idea to have an architecture field here.

Reply:
Ahh, good points, thanks. Look forward to more feedback. Arch does make more sense than target.

Reply:
And has an example of architecture, I would of recommend using "psa"